sys/random/random.c

```diff
@@ -54,3 +54,27 @@ void random_bytes(uint8_t *target, size_t n)
         *target++ = *random_pos++;
     }
 }
+
+uint32_t random_uint32_range(uint32_t a, uint32_t b)
+{
+    uint32_t range = b - a;
+    uint32_t highest_pow2 = (1 << bitarithm_msb(range));
+    uint32_t random = random_uint32();
+
+    /* check if range is a power of two */
+    if (!(range % highest_pow2)) {
```

Comment:
the compiler won't be able to optimize the modulo operation. To check whether `range` is a power of two, you could use
```c
range && !( range & ( range - 1))
```